var/spack/repos/builtin/packages/openspeedshop/package.py

```diff
@@ -130,84 +139,220 @@ class Openspeedshop(Package):
     depends_on("cbtf", when='+cbtf')
     depends_on("cbtf-krell", when='+cbtf')
     depends_on("cbtf-argonavis", when='+cbtf+cuda')
-    depends_on("mrnet@5.0.1:+lwthreads+krellpatch", when='+cbtf')
+    depends_on("mrnet@5.0.1:+lwthreads", when='+cbtf')
+
+    # We don't need this for building, but to complete the
+    # runtime module file
+    # It is required in cbtf, so it will be available and not
+    # cause another rebuild
+
+    depends_on("xerces-c@3.1.1:", when='+cbtf')
 
     def adjustBuildTypeParams_cmakeOptions(self, spec, cmakeOptions):
-        # Sets build type parameters into cmakeOptions the options that will
-        # enable the cbtf-krell built type settings
+        # Sets build type parameters into cmakeOptions the
+        # options that will enable the cbtf-krell built type settings
 
         compile_flags = "-O2 -g"
         BuildTypeOptions = []
-        # Set CMAKE_BUILD_TYPE to what cbtf-krell wants it to be, not the
-        # stdcmakeargs
+        # Set CMAKE_BUILD_TYPE to what cbtf-krell wants it
+        # to be, not the stdcmakeargs
         for word in cmakeOptions[:]:
             if word.startswith('-DCMAKE_BUILD_TYPE'):
                 cmakeOptions.remove(word)
             if word.startswith('-DCMAKE_CXX_FLAGS'):
                 cmakeOptions.remove(word)
             if word.startswith('-DCMAKE_C_FLAGS'):
                 cmakeOptions.remove(word)
-        BuildTypeOptions.extend([
-            '-DCMAKE_BUILD_TYPE=None',
-            '-DCMAKE_CXX_FLAGS=%s'         % compile_flags,
-            '-DCMAKE_C_FLAGS=%s'           % compile_flags
-        ])
+        BuildTypeOptions.extend(['-DCMAKE_BUILD_TYPE=None',
+                                 '-DCMAKE_CXX_FLAGS=%s'  % compile_flags,
+                                 '-DCMAKE_C_FLAGS=%s'    % compile_flags])
 
         cmakeOptions.extend(BuildTypeOptions)
 
     def set_mpi_cmakeOptions(self, spec, cmakeOptions):
-        # Appends to cmakeOptions the options that will enable the appropriate
-        # MPI implementations
+        # Appends to cmakeOptions the options that will enable
+        # the appropriate MPI implementations
 
         MPIOptions = []
 
         # openmpi
         if '+openmpi' in spec:
-            MPIOptions.extend([
-                '-DOPENMPI_DIR=%s' % spec['openmpi'].prefix
-            ])
+            MPIOptions.extend(['-DOPENMPI_DIR=%s' % spec['openmpi'].prefix])
```

Comment:
Each of the MPI checks could add to the element list.  OSS supports multiple MPI implementations in one build.